tools/batchTestGenerator/internal/generateBatchValues.go

```diff
@@ -0,0 +1,18 @@
+package internal
+
+import (
+	"fmt"
+	"strings"
+)
+
+type testSetInput []TestCaseInfo
+
+func generateBatchValues(testSet testSetInput) (string, error) {
+	var formattedTestCase []string
+
+	for _, tsi := range testSet {
+		currentTestCase := fmt.Sprintf("%s %s %s", tsi.serviceType, tsi.testcaseName, tsi.additionalVar)
+		formattedTestCase = append(formattedTestCase, currentTestCase)
+	}
+	return strings.Join(formattedTestCase, "\n"), nil
```

Comment:
```suggestion
	var b strings.Builder

	for _, tsi := range testSet {
		fmt.Fprintf(&b, "%s %s %s\n", tsi.serviceType, tsi.testcaseName, tsi.additionalVar)
	}

	return b.String(), nil
```

A `strings.Builder` is probably the way to go here, no need to build up a string slice to join later.